ori/volume/server/volume_delete.go

```diff
@@ -18,27 +18,58 @@ import (
 	"context"
 	"fmt"
 
+	"github.com/go-logr/logr"
 	ori "github.com/onmetal/onmetal-api/ori/apis/volume/v1alpha1"
-	"google.golang.org/grpc/codes"
-	"google.golang.org/grpc/status"
-	apierrors "k8s.io/apimachinery/pkg/api/errors"
 )
 
-func (s *Server) DeleteVolume(ctx context.Context, req *ori.DeleteVolumeRequest) (*ori.DeleteVolumeResponse, error) {
-	volumeID := req.VolumeId
-	log := s.loggerFrom(ctx, "VolumeID", volumeID)
-
-	onmetalVolume, err := s.getAggregateCephVolume(ctx, req.VolumeId)
+func (s *Server) findNameFromId(ctx context.Context, volumeId string) (string, bool, error) {
+	mappings, err := s.provisioner.GetAllMappings(ctx, RbdImage)
 	if err != nil {
-		return nil, err
+		return "", false, fmt.Errorf("unable to fetch all volume mapping: %w", err)
 	}
 
-	log.V(1).Info("Deleting volume")
-	if err := s.client.Delete(ctx, onmetalVolume.Pvc); err != nil {
-		if !apierrors.IsNotFound(err) {
-			return nil, fmt.Errorf("error deleting onmetal volume: %w", err)
+	for volumeName, imageId := range mappings {
+		if volumeId == imageId {
+			return volumeName, true, nil
 		}
-		return nil, status.Errorf(codes.NotFound, "volume %s not found", volumeID)
+	}
+
+	return "", false, nil
+}
+
+func (s *Server) deleteCephImage(ctx context.Context, log logr.Logger, volumeId string) (retErr error) {
+
+	volumeName, found, err := s.findNameFromId(ctx, volumeId)
+	if err != nil {
+		return fmt.Errorf("unable to find volumeName from id: %w", err)
+	}
+	if !found {
+		log.V(1).Info("No mapping found: Already gone.", "imageId", volumeId)
+		return nil
+	}
+
+	log.V(2).Info("Try to acquire lock for volume", "volumeName", volumeName)
+	if err := s.Lock(volumeName); err != nil {
+		return fmt.Errorf("unable to acquire lock: %w", err)
+	}
+	defer s.Release(volumeName)
```

Comment:
`Lock`ing / `Release`ing should be done at the server level.

Reply:
`s.Lock`/`s.Release` operates on the server